ui/src/app/components/toast-banner.tsx

```diff
@@ -61,40 +60,42 @@ export enum ToastType {
   WARNING
 }
 
-const styleForType = (tt: ToastType) => switchCase(tt,
-  [ToastType.INFO, () => styles.infoBanner],
+const styleForType = (toastType: ToastType, zIndex): React.CSSProperties => switchCase(toastType,
+  [ToastType.INFO, () => ({
+    ...styles.infoBanner,
+    zIndex
+  })],
   [ToastType.WARNING, () => ({
-      ...styles.infoBanner,
-      backgroundColor: colorWithWhiteness(colors.highlight, .5),
+    ...styles.infoBanner,
+    zIndex,
+    backgroundColor: colorWithWhiteness(colors.highlight, .5),
   })],
 );
 
 interface ToastProps {
   title: string;
-  message: string;
+  message: string | JSX.Element;
   onClose: Function;
-  type: ToastType;
-  footer?: JSX.Element;
-}
-
-const InnerComponent = (props: ToastProps) => {
-  const {title, message, onClose, type, footer} = props;
-  return <FlexColumn style={styleForType(type)}>
-    <FlexRow style={{alignItems: 'center', marginTop: '.1rem'}}>
-      {type === ToastType.WARNING && warningIcon}
-      <div style={styles.title}>{title}</div>
-    </FlexRow>
-    <div style={styles.message}>{message}</div>
-    <div style={styles.footer}>{footer}</div>
-    <ClrIcon
-      shape={'times'}
-      size={20}
-      style={styles.closeIcon}
-      onClick={() => onClose()}
-    />
-  </FlexColumn>;
+  toastType: ToastType;
+  zIndex: any;  // TODO better type
+  footer?: string | JSX.Element;
 }
-
 export const ToastBanner = (props: ToastProps) => {
-  return ReactDOM.createPortal(<InnerComponent {...props}/>, document.getElementsByTagName('body')[0]);
+  const {title, message, onClose, toastType, zIndex, footer} = props;
+  return ReactDOM.createPortal(
+    <FlexColumn style={styleForType(toastType, zIndex)}>
+      <FlexRow style={{alignItems: 'center', marginTop: '.1rem'}}>
+        {toastType === ToastType.WARNING && warningIcon}
+        <div style={styles.title}>{title}</div>
+      </FlexRow>
+      <div style={styles.message}>{message}</div>
+      <div style={styles.footer}>{footer}</div>
+      <ClrIcon
```

Comment:
nit: why not the other icon type

Reply:
this is the way it was originally.  I don't know enough about icons to have a preference.  Is there a better choice here?

Reply:
I was thinking about consistency across our code, since for all our new one we are using the font awesome icons i assumed we are going to change. 
If it was there originally  and we need to think more about which icons to use when (which i think we all as a team do ) we do not have to tackle this icon change in this PR

Reply:
I think that would be a good topic for an Eng Sync